Sources/URLNavigator.swift

```diff
@@ -179,6 +180,42 @@ open class URLNavigator {
     return viewController
   }
 
+  /// Pushes a view controller using `UINavigationController.pushViewController()`.
+  ///
+  /// This is an example of pushing a view controller to the top-most view contoller:
+  ///
+  ///     Navigator.push("myapp://user/123")
+  ///
+  /// Use the return value to access a view controller.
+  ///
+  ///     let userViewController = Navigator.push("myapp://user/123")
+  ///     userViewController?.doSomething()
+  ///
+  /// Add extra `parameters: [String: Any]`:
+  ///
+  ///     let userViewController = Navigator.push("myapp://user/123", ["callBackBlock": YourHandler, "object": YourModel])
+  ///
+  /// - parameter url: The URL to find view controllers.
+  /// - parameter parameters: The parameters that you want to added.
+  /// - parameter from: The navigation controller which is used to push a view controller. Use application's top-most
+  ///     view controller if `nil` is specified. `nil` by default.
+  /// - parameter animated: Whether animates view controller transition or not. `true` by default.
+  ///
+  /// - returns: The pushed view controller. Returns `nil` if there's no matching view controller or failed to push
+  ///            a view controller.
+  @discardableResult
+  open func push(
+    _ url: URLConvertible,
+    _ userInfo: [AnyHashable: Any]?,
```

Comment:
Why not just adding `userInfo` parameter with default value (`nil`) to existing `push()` and `present()` function?